lib/node_modules/@stdlib/blas/base/srotg/lib/assign.js

```diff
@@ -0,0 +1,94 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2023 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var abs = require( '@stdlib/math/base/special/abs' );
+var sqrt = require( '@stdlib/math/base/special/sqrt' );
+var pow = require( '@stdlib/math/base/special/pow' );
+var copysign = require( '@stdlib/math/base/special/copysign' );
+
+
+// MAIN //
+
+/**
+* Constructs a Givens plane rotation.
+*
+* @param {number} a - rotational elimination parameter
+* @param {number} b - rotational elimination parameter
+* @param {Float32Array} out - destination array
+* @param {integer} stride - index increment
+* @param {NonNegativeInteger} offset - starting index
+* @returns {Float32Array} destination array
+*
+* @example
+* var Float32Array = require( '@stdlib/array/float32' );
+*
+* var out = srotg( 0.0, 2.0, new Float32Array( 4 ), 1, 0 );
+*  // returns <Float32Array>[ 2.0, 1.0, 0.0, 1.0 ]
+*/
+function srotg( a, b, out, stride, offset ) {
+	var scale;
+	var sign;
+	var aa;
+	var ab;
+	var r;
+	var c;
+	var s;
+	var z;
+
+	aa = abs( a );
+	ab = abs( b );
+	if ( aa > ab ) {
+		sign = copysign( 1.0, a );
+	} else {
+		sign = copysign( 1.0, b );
+	}
+	scale = aa + ab;
```

Comment:
Just realized this, but we should be emulating single-precision floating-point arithmetic here. E.g.,

```javascript
scale = float64ToFloat32( aa + ab );
```

This applies here and the result of each operation.

Reply:
Currently, operations are being performed in "extended" precision. In order to match C, we need to "convert" the result of each operation to a `float32`.